Symfony/CS/Tests/FileCacheManagerIntegrationTest.php

```diff
@@ -0,0 +1,129 @@
+            ),
+            'two' => array(
+                'filename' => $this->rootDir.DIRECTORY_SEPARATOR.'path_two'.DIRECTORY_SEPARATOR.'CommonFilename.php',
+                'content' => "<?\n// two\n",
+            ),
+        );
+
+        $this->fs->mkdir($this->rootDir);
+        $this->fs->mkdir($this->customConfigDir);
+
+        array_map(function (array $file) {
+            $this->fs->mkdir(dirname($file['filename']));
+            file_put_contents($file['filename'], $file['content']);
+        }, $this->files);
+    }
+
+    protected function tearDown()
+    {
+        $this->clearFileCacheManagerIntegrationTestFiles();
+    }
+
+    private function clearFileCacheManagerIntegrationTestFiles()
+    {
+        $this->fs->remove($this->rootDir);
+    }
+
+    private function isCacheAvailable()
+    {
+        $fileCacheManager = new FileCacheManager(true, $this->rootDir, array());
+        $reflectionIsCacheAvailable = new \ReflectionMethod($fileCacheManager, 'isCacheAvailable');
+        $reflectionIsCacheAvailable->setAccessible(true);
+        $isCacheAvailable = $reflectionIsCacheAvailable->invoke($fileCacheManager);
+        $reflectionIsCacheAvailable->setAccessible(false);
+
+        // We don't want FileCacheManager::__destruct() interferences
+        unset($reflectionIsCacheAvailable, $fileCacheManager);
+
+        return $isCacheAvailable;
+    }
+
+    /**
+     * @dataProvider provideScenarios
+     */
+    public function testCacheScenarios($insideCacheFileDir)
+    {
+        if (!$this->isCacheAvailable) {
+            $this->markTestSkipped('Cache functionality not supported');
+        }
+
+        $configDir = $this->rootDir;
+        if (!$insideCacheFileDir) {
+            $configDir = $this->customConfigDir;
+        }
+
+        $fixer = new Fixer();
+        $fixer->addFixer(new Fixer\PSR1\ShortTagFixer());
+        $fixers = $fixer->getFixers();
+
+        $files = array_map(function ($file) {
+            return new FinderSplFileInfo(
+                $file['filename'],
+                dirname($file['filename']),
+                basename($file['filename'])
+            );
+        }, $this->files);
+
+        $fileCacheManager = new FileCacheManager(true, $configDir, $fixers);
+        $this->assertNotNull($fixer->fixFile($files['one'], $fixers, false, false, $fileCacheManager));
+        $this->assertNotNull($fixer->fixFile($files['two'], $fixers, false, false, $fileCacheManager));
+
+        // This unset() calls FileCacheManager::saveToFile() and is needed for the test
+        unset($fileCacheManager);
+
+        $fileCacheManager = new FileCacheManager(true, $configDir, $fixers);
+        $this->assertFalse($fileCacheManager->needFixing($files['one']->getRealpath(), file_get_contents($files['one']->getRealpath())));
+        $this->assertFalse($fileCacheManager->needFixing($files['two']->getRealpath(), file_get_contents($files['two']->getRealpath())));
```

Comment:
I consider this very dangerous: if for any reason in the future the `$files` variable will result empty, the test will pass silently because no assertion will be done.

It happened some times in our business and in a bunch of OSS projects: I strongly discourage dynamic assertions.